scipy/linalg/_backend.py

```diff
@@ -0,0 +1,60 @@
+import scipy._lib.uarray as ua
+from scipy.linalg import _api
+
+
+__all__ = ['register_backend', 'set_backend',
+           'set_global_backend', 'skip_backend']
+
+
+class _ScipyLinalgBackend:
+    __ua_domain__ = "numpy.scipy.linalg"
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = getattr(_api, method.__name__, None)
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+
+_named_backends = {
+    'scipy': _ScipyLinalgBackend,
+}
+
+
+def _backend_from_arg(backend):
+
+    if isinstance(backend, str):
+        try:
+            backend = _named_backends[backend]
+        except KeyError as e:
+            raise ValueError('Unknown backend {}'.format(backend)) from e
+
+    if backend.__ua_domain__ != 'numpy.scipy.linalg':
+        raise ValueError('Backend does not implement "numpy.scipy.linalg"')
+
+    return backend
+
+
+def set_global_backend(backend, coerce=False, only=False, try_last=False):
+    backend = _backend_from_arg(backend)
+    ua.set_global_backend(backend, coerce=coerce, only=only, try_last=try_last)
+
+
+def register_backend(backend):
+    backend = _backend_from_arg(backend)
+    ua.register_backend(backend)
+
+
+def set_backend(backend, coerce=False, only=False):
+    backend = _backend_from_arg(backend)
+    return ua.set_backend(backend, coerce=coerce, only=only)
+
+
+def skip_backend(backend):
```

Comment:
Why should anyone skip any backend? I don't get any of these method names.

Reply:
It allows for meta-backends, e.g.
```python
class LoggingBackend:
    __ua_domain__ = 'numpy.scipy'

    def __ua_function__(self, method, args, kwargs):
        with skip_backend(self):
           res =  method(*args, **kwargs)
        logging.info(f'__ua_function__({method}, {args}, {kwargs}) -> {ret}')
        return res
```

You call a function once and it invokes the `LoggingBackend`, then it re-dispatches to the next viable registered backend.

Reply:
I still don't see which backend is skipped here. This is almost what `partial` from `functools` does.

Reply:
This is nothing like `functools.partial`. It's more like being able to add/remove function decorators at runtime.

Reply:
Yes I'm giving an functools example from a naming point of view not functionality. You are adding/removing decorators not skipping anything.

Reply:
>Yes I'm giving an functools example from a naming point of view not functionality.

I don't understand what you mean by this: do you just mean `functools.partial` and `ua.skip_backend` are badly named? Not that they are somehow similar in functionality.

>You are adding/removing decorators not skipping anything.

`uarray` maintains an specific order in which backends are tried during dispatch, so if you make the same call twice then the same backends are tried in the same order. When you skip a backend it gets added to a separate list of skipped backends. 
So, it's still registered but gets skipped over during dispatch because it's in the list of skipped backends.

Reply:
xref Quansight-Labs/uarray#237, in particular https://github.com/Quansight-Labs/uarray/issues/237#issuecomment-613683216

We also moved to the same approach in `unumpy`.